trace_decoder/src/lib.rs

```diff
@@ -1,14 +1,141 @@
+//! This library is intended to genrate an Intermediary Representation (IR) of a
+//! block's transactions, given a [BlockTrace] and some additional
+//! data [OtherBlockData].
+//!
+//! A [BlockTrace] is defined as follows:
+//! ```ignore
+//! pub struct BlockTrace {
+//!     /// The trie pre-images (state and storage) in multiple possible formats.
+//!     pub trie_pre_images: BlockTraceTriePreImages,
```

Comment:
Are state and storage mashed together into one trie?

I have to admit that I don't quite understand what these are. Should I understand this as the "actual data" that will be hashed and put into tries later in the process? It's the word "pre-image" that trips me up here, especially given the text later mentions `HashedPartialTries`, suggesting the data is already hashed? Can you elaborate here in a reasonably terse way?

Reply:
From my understanding of the library, the pre-images are the tries before the start of the block execution. I will add this to the description.